.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,157 @@
+        default: 'vx.x.x'
+        required: true
+        type: string
+      base_branch:
+        description: "The base branch to release from; only override this if preparing a release off of the non-default branch"
+        default: 'main'
+        required: true
+        type: string
+
+env:
+  RELEASE_VERSION: ${{ inputs.release_version }}
+  RELEASE_BRANCH: release/${{ inputs.release_version }}
+
+jobs:
+  create-release-pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set Release Version and Branch to Check Out
+        id: set-release
+        run: |
+          if [[ $RELEASE_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "release-tag: $RELEASE_VERSION"
+            echo "release-tag=$RELEASE_VERSION" >> $GITHUB_OUTPUT
+          else
+            echo "Version input doesn't match the regex pattern ^[0-9]+\.[0-9]+\.[0-9]+$"
+            exit 1
+          fi
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.base_branch }}
+
+      - name: Create Release Branch if it does not exist
+        run: |
+          if ! git show-ref --verify --quiet "refs/remotes/origin/$RELEASE_BRANCH"; then
+            git checkout -b $RELEASE_BRANCH
+            git push --set-upstream origin $RELEASE_BRANCH
+          elif [[ $(git rev-parse --abbrev-ref HEAD) != "$RELEASE_BRANCH" ]]; then
+            echo "Current Branch: $(git rev-parse --abbrev-ref HEAD)"
+            echo "Release branch exists, make sure you're using the workflow from the release branch or delete the existing release branch."
+            exit 1
+          else
+            echo "Release branch exists and used as workflow ref."
+          fi
+
+      - name: Get Latest Release
+        id: get-release
+        run: |
+          echo "Get the latest stable release"
+          tag=$(curl -L \
+                      --header "Accept: application/vnd.github.v3+json" \
+                      "https://api.github.com/repos/${{ github.repository }}/releases/latest" | jq -r '.tag_name')
+          echo "latest-tag=$tag" >> $GITHUB_OUTPUT
+
+
+      - name: Build Changelog
+        id: build-changelog
+        env:
+          PREVIOUS_VERSION: ${{ steps.get-release.outputs.latest-tag }}
+          RELEASE_TAG: ${{ steps.set-release.outputs.release-tag }}
+        run: |
+          CHANGELOG=$(curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ github.repository }}/releases/generate-notes \
+            -d '{"tag_name":"${{ env.RELEASE_TAG }}","target_commitish":"${{ env.RELEASE_BRANCH }}","previous_tag_name":"${{ env.PREVIOUS_VERSION }}","configuration_file_path":".github/create-release-pr.yml"}' \
```

Comment:
```suggestion
            -d '{"tag_name":"${{ env.RELEASE_TAG }}","target_commitish":"${{ env.RELEASE_BRANCH }}","previous_tag_name":"${{ env.PREVIOUS_VERSION }}","configuration_file_path":".github/release.yml"}' \
```

Reply:
I changed this intentionally - I believe it's only effect is spitting out an incorrect source for the generation in the CHANGELOG (the source is actually create-release-pr.yml).

Reply:
Oh weird. In that case it should probably be `.github/workflow/create-release-pr.yml` (note the missing `workflow`).

Reply:
You are right, I am wrong - taking your change.